send_review_notification/github_services.py

```
@@ -0,0 +1,259 @@
+                discussionCategories(first: 10) {
+                    nodes {
+                        id
+                        name
+                        repository {
+                            id
+                            discussions(last: 10) {
+                                edges {
+                                    node {
+                                        id
+                                        title
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    """
+
+    variables = {
+        'org_name': org_name,
+        'repository': repo
+    }
+
+    response = requests.post(
+        GITHUB_GRAPHQL_URL,
+        json={'query': query, 'variables': variables},
+        headers=_get_request_headers()
+    )
+
+    data = response.json()
+
+    discussion_id = ''
+    category_name = 'Reviewer notifications'
+    discussion_title = 'Pending reviews'
+    discussion_categories = (
+        data['data']['repository']['discussionCategories']['nodes'])
+    try:
+        for category in discussion_categories:
+            if category['name'] == category_name:
+                try:
+                    discussions = category['repository']['discussions']['edges']
+                    for discussion in discussions:
+                        if discussion['node']['title'] == discussion_title:
+                            discussion_id = discussion['node']['id']
+                            break
+                except Exception as e:
+                    raise Exception(
+                        'Discussion with title %s not found, please create'
+                        'a discussion with that title.' % (
+                        discussion_title)) from e
+    except Exception as e:
+        raise Exception('%s category is missing in GitHub Discussion.' % (
+            category_name)) from e
```

I don't think the try...except blocks here will do what you want. For the outer try...except, I think you want to throw an exception if there is no category where `category['name'] == 'Reviewer notifications'`. When this is the case, you'll loop through all the categories in `discussion_categories`. For each category, `category['name'] == category_name` will be False, so you'll never enter the `if` block, but no exception that would trigger the try...except will be thrown. The same problem exists for the inner try...except

Instead, I think you want to check whether the `if` block executes, and if it doesn't, raise an exception

---

I have updated this part and after trying with the wrong date seems like its working. PTAL